attestation-service/verifier/src/az_snp_vtpm/mod.rs

```diff
@@ -34,20 +34,25 @@ pub struct AzSnpVtpm;
 impl Verifier for AzSnpVtpm {
     async fn evaluate(
         &self,
-        nonce: String,
-        attestation: &Attestation,
+        evidence: &[u8],
+        report_data: Option<&[u8]>,
+        init_data_hash: Option<&[u8]>,
     ) -> Result<TeeEvidenceParsedClaim> {
-        let evidence = serde_json::from_str::<Evidence>(&attestation.tee_evidence)
+        if report_data.is_none() {
+            bail!("Azure SNP vTPM verifier must provide report data field!");
+        }
```

Comment:
@mkulke Hi Magnus, I've changed the API of verifier of `AzSnpVtpm`. PTL if it looks good to you.

BTW, it will now change the behavior of current CoCoAS.